projects/uriparser/uri_dissect_query_malloc_fuzzer.cc

```diff
@@ -42,17 +42,20 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
   if (query_list == nullptr || result != URI_SUCCESS || item_count < 0)
     return 0;
 
-  int chars_required;
+  int chars_required = 0;
   if (uriComposeQueryCharsRequiredA(query_list, &chars_required) != URI_SUCCESS)
     return 0;
-    
+
+  if (!chars_required) {
```

Comment:
if i understand the documentation correctly, chars_required should be incremented by 1, to account for the \0 terminator.
uriComposeQueryCharsRequiredA "returns" the number of chars EXCLUDING the \0 terminator.
uriComposeQueryA expects the size of the buffer INCLUDING the \0 terminator.
this way you never give it a NULL buffer.
but i'd argue, that it is still a bug in uriComposeQueryA if it accesses the buffer when chars_required is 0.